src/java/org/apache/cassandra/audit/AuditLogManager.java

```diff
@@ -197,8 +198,38 @@ public void logBatch(String batchTypeName, List<Object> queryOrIdList, List<List
         }
     }
 
-    public void reloadFilters()
+    public synchronized void disableAuditLog()
+    {
+        logger.info("Audit logging is disabled, stopping any existing loggers"); //debug
+        if (this.auditLogger != null)
+        {
+            this.auditLogger.stop();
+            this.auditLogger = null;
+        }
+    }
+
+    public synchronized void enableAuditLog(AuditLogOptions auditLogOptions) throws ConfigurationException
     {
-        filter = AuditLogFilter.create();
+        logger.trace("Audit logging is being enabled. Reloading AuditLogOptions.");
+        IAuditLogger oldLogger = this.auditLogger;
+        filter = AuditLogFilter.create(auditLogOptions);
+
+        if (oldLogger != null && oldLogger.getClass().getSimpleName().equals(auditLogOptions.logger))
+        {
+            logger.info("New AuditLogger [{}] is same as existing logger, hence not initializing the logger", auditLogOptions.logger);
+            return;
+        }
+
+        this.auditLogger = getAuditLogger(auditLogOptions.logger);
+
+            /* Ensure oldLogger's stop() is called after we swap it with new logger otherwise,
+             * we might be calling log() on the stopped logger.
+             */
+
+        if (oldLogger != null)
+        {
+            oldLogger.stop();
```

Comment:
I'm somewhat concerned about the race conditions inherent in this for the ``BinAuditLogger``, for the ``FileAuditLogger`` it seems clear this would be ok. In particular I wonder if ``log`` could raise all sorts of ``RuntimeExceptions`` if ``oldLogger.log`` get's called in ``log`` after this thread has called ``stop``. This may be fine, but just curious if we can at least add a comment here explaining the order of these operations and how that might impact live queries.

Reply:
As we discussed offline, BinAuditLogger also handles such cases.